### PR TITLE
Fix BroadcastToSequence to enable context features in sequential models

### LINKED ISSUE #989: [BUG] BroadcastToSequence class does not work properly to broadcast the scalar inputs to list inputs
### Bug description

When I use `post = mm.BroadcastToSequence(context_schema, seq_schema),` to broadcast a continuous scalar item as a list column, I noticed that the input block does not include my scalar item.. Looks like ` mm.BroadcastToSequence()` is not working as expected. 

### Steps/Code to reproduce bug

Please run the code below to reproduce the issue/error:

```
from merlin.schema.tags import Tags
from merlin.io.dataset import Dataset
from merlin.datasets.synthetic import generate_data
train = generate_data("sequence-testing", num_rows=1000)
seq_schema =train.schema.select_by_name(['item_id_seq', 'categories', 'item_age_days_norm'])
context_schema = train.schema.select_by_name(['user_age'])
train.schema = train.schema.select_by_name(['item_id_seq', 'categories', 'item_age_days_norm', 'user_age'])

target_schema = train.schema.select_by_tag(Tags.ITEM_ID)
target = target_schema.column_names[0]
print(target)

item_id_name = train.schema.select_by_tag(Tags.ITEM_ID).first.properties['domain']['name']
dmodel =32
manual_dims = {
    'item_id_seq': dmodel, 
    'categories': 16
}

mlp_block = mm.MLPBlock(
                [128,dmodel],
                activation='relu',
                no_activation_last_layer=True,
)


input_block = mm.InputBlockV2(
        context_schema + seq_schema,    # if we only feed seq_schema it works but it wont consider the broadcasted feature.
        embeddings=mm.Embeddings(
            seq_schema.select_by_tag(Tags.CATEGORICAL), 
            sequence_combiner=None,
            dim=manual_dims
        ),
        post = mm.BroadcastToSequence(context_schema, seq_schema),
        aggregation=None
)

dense_block =mm.SequentialBlock(
    input_block,
    mlp_block,
    mm.XLNetBlock(d_model=dmodel, n_head=4, n_layer=2, 
                   pre=mm.ReplaceMaskedEmbeddings(),
                   post="inference_hidden_state",
                  )
)

mlp_block2 = mm.MLPBlock(
                [128,dmodel],
                activation='relu',
                no_activation_last_layer=True,
)

prediction_task= mm.CategoricalOutput(
    to_call=input_block["categorical"][item_id_name],
)
model_transformer = mm.Model(dense_block, mlp_block2, prediction_task)

model_transformer.compile(run_eagerly=False, optimizer='adam', loss="categorical_crossentropy",
              metrics=mm.TopKMetricsAggregator.default_metrics(top_ks=[4])
             )
model_transformer.fit(train, batch_size=512, epochs=1, pre=mm.SequenceMaskRandom(schema=seq_schema, target=target, masking_prob=0.3))
```
Then check the input block output with a sample batch

```
batch= mm.sample_batch(train, batch_size=100, include_targets=False, to_ragged=True)
input_block(batch).keys()
```

you can see that it does not print out the `user_age` which is the context feature that is supposed to be broadcasted. Similarly if you print `model_transformer` you will see that the model architecture does not have `user_age` feature in any layer.

PLEASE NOTE THAT IF I feed `train.schema` to input block instead of `seq_schema` I get an error. How should I make this work?

### Expected behavior
The `BroadcastToSequence` should work as expected.

### Environment details
- Merlin version: 
- Platform:
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

I am using `merlin-tensorflow:22.12` with the latest main branches pulled.

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
Fixes #989

### Goals :soccer:
This PR fixes `BroadcastToSequence`, which was not working properly in some cases

### Implementation Details :construction:
- `BroadcastToSequence` was not working in graph mode because the contextual features were being expanded to match the shape of sequential features using a logic like the following snippet, which was making the first dim (batch size) of the resulting tensor fixed, while the other sequential tensors had the first dim as None (bcs the batch size might vary).
```python
sequence_length = inputs["a_sequential_feature"].row_lengths()
broadcasted_contextual_feature = tf.RaggedTensor.from_row_lengths(
    tf.repeat(inputs["a_contextual_feature"], sequence_length, axis=0), 
    sequence_length 
)
```

The solution I found to keep the first dim of the expanded context feature tensor matching the other sequential features was using `tf.ones_like()` to create a (ragged) tensor matching the sequential feature shape and then multiply by the context feature (equivalent to repeat it), as the following example

```python
first_seq_feature_name = list(seq_features_shapes.keys())[0]
non_seq_target[fname] = tf.ones_like(
                            inputs[first_seq_feature_name][..., :1]
                         ) * tf.expand_dims(inputs["a_contextual_feature"], -1)
```

This PR also includes other fixes:
- `Continuous` block now forwards masking (`supports_masking=True`)
- `BroadcastToSequence` was refactored to separate the logic that checks if sequential and context features in the input match the schema, otherwise exceptions are raised now.
- In `BroadcastToSequence.compute_mask()`  I don't call `self._broadcast()` and the logic there was made simpler: the expanded contextual feature mask should match the sequential feature mask.
- The `SequenceTargetAsInput` now returns a `tuple` instead of a `Prediction`, so that Keras can better align the `inputs` and `targets` output from `call()` and `compute_mask()` of child classes.
- In `ProcessList`, reshaping scalar features (is_list=False,is_ragged=False) to be 2D (batch size, 1), as the last dim was None in graph mode and was causing issues when concatenating

### Testing Details :mag:
Have added new tests for testing `BroadcastToSequence` usage as a `post` of `InputBlockV2`, with both categorical and continuous context (non-sequential) features and also to test it in a Transformer model trained with masked language modeling.

### CODE REVIEW

File: merlin/models/tf/transforms/tensor.py [hunk 1]
```diff
@@ -101,6 +101,13 @@ def call(self, inputs: TabularData, **kwargs) -> TabularData:
             elif isinstance(val, tf.RaggedTensor):
                 ragged = val
             else:
+                # Expanding / setting last dim of non-list features to be 1D
```
Comment (oliverholworthy) [inline merlin/models/tf/transforms/tensor.py:104]: is this relevant to `ProcessList`. intuitively `ProcessList` sounds like it might only be transforming list features

Comment (gabrielspmoreira) [reply merlin/models/tf/transforms/tensor.py:104]: That is a good point @oliverholworthy . The `ProcessList` is currently used at [core places](https://github.com/NVIDIA-Merlin/models/search?q=process_list) to ensure the features are in good shape for models. When the change making dataloader outputs scalars as 1D happens, we will also need this fix that makes scalars 2D (batch size, 1) for models.
What if we rename `ProcessList` to `PrepareFeatures` and have it as a generic block that works as a translation layer between dataloader and models (used in the same [places](https://github.com/NVIDIA-Merlin/models/search?q=process_list) `ProcessList` is currently used)?

**Review by marcromeyn — APPROVED** [review]

File: merlin/models/tf/transforms/tensor.py (hunk 1 above)
Comment (oliverholworthy) [reply merlin/models/tf/transforms/tensor.py:104]: The rename sounds like a reasonable thing to do and better matches it's purpose. Can be in another PR if preferred

**Review by oliverholworthy — APPROVED** [review]

File: merlin/models/tf/transforms/tensor.py (hunk 1 above)
Comment (gabrielspmoreira) [reply merlin/models/tf/transforms/tensor.py:104]: @oliverholworthy I have created another separate PR just for renaming `ProcessList` to `PrepareFeatures`: PR #992
